src/modal/component/accordion/button/AccordionButton.tsx

```diff
@@ -0,0 +1,22 @@
+import "./_accordion-button.scss";
+
+import React from "react";
+
+import AccordionIcon from "../icon/AccordionIcon";
+
+interface AccordionButtonProps {
+  children: React.ReactNode;
+  onClick: VoidFunction;
+}
+
+function AccordionButton({children, onClick}: AccordionButtonProps) {
+  return (
+    <button className={"accordion-button"} onClick={onClick}>
+      <AccordionIcon />
```

Comment:
I think the project will not be large so much and there is no any advantage to extract this as a component here I believe.

I think the same for `AccordionButton` , `AccordionItem` and `AccordionPanel`.
Also we can pass accordion types as AccordionProps interface inside the `Accordion.tsx`.
I don't know the others but It might be better for readability to have  Accordion as a complete component by itself.

Preventing flat folder structure as much as we can increases the readability in my opinion.
Of course we can discuss this issue further ✌🏼

Reply:
Got your point @akgunberk but it doesn't affect the output not much so I'd prefer to keep those separate.